pkg/syncer/metadatasyncer.go

```diff
@@ -3196,3 +3228,103 @@ func storagePolicyUsageCRSync(ctx context.Context, metadataSyncer *metadataSyncI
 		}
 	}
 }
+
+// startAvailabilityZoneCRInformer listens on changes to AvailabilityZone CR instances
+func startAvailabilityZoneCRInformer(ctx context.Context, cfg *restclient.Config) error {
+	log := logger.GetLogger(ctx)
+	// Check if AZ CR is registered in the environment.
+	// Create a new AvailabilityZone client.
+	azClient, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to create AvailabilityZone client using config. Err: %+v", err)
+	}
+	// Get AvailabilityZone list
+	azResource := schema.GroupVersionResource{
+		Group: availabilityZoneCRGroupName, Version: availabilityZoneCRVersion,
+		Resource: availabilityZoneCRResourceName}
+	_, err = azClient.Resource(azResource).List(ctx, metav1.ListOptions{})
+	// Handling the scenario where AvailabilityZone CR is not registered in the
+	// supervisor cluster.
+	if apiMeta.IsNoMatchError(err) {
+		log.Info("AvailabilityZone CR is not registered on the cluster")
+		return common.ErrAvailabilityZoneCRNotRegistered
+	}
```

Comment:
Is this possible? Can the latest CSI driver run on old supervisor clusters? If it's an older supervisor cluster, then wcpsvc will probably not create the AZ cr. This is unlikely, but can you check on this CSI driver-supervisor compatibility?

Reply:
This is unlikely case. But on a safer side I tried to follow same thing which is implemented in topology code. We are just silently logging error in case AZ CR is not registered.

Reply:
Our code is under storage quota M2 FSS, so it will never run on old supervisor cluster. So, we should not face this issue.
In any case if we hit this issue, then I am erroring out.